src/recipe.rs

```diff
@@ -133,11 +133,24 @@ impl<'src, D> Recipe<'src, D> {
   }
 
   fn working_directory<'a>(&'a self, context: &'a ExecutionContext) -> Option<PathBuf> {
-    if self.change_directory() {
-      Some(context.working_directory())
-    } else {
-      None
+    if !self.change_directory() {
+      return None;
     }
+
+    let working_dir = self
```

Comment:
I think this is better implemented as a for loop:

```rust
let working_directory = context.working_directory();

for attribute in self.attributes {
  if let Attribute::WorkingDirectory(dir) = attribute {
    return Some(working_directory.join(dir.cooked);
  }
}

Some(working_directory)
```

Also note the use of `cooked`, which is the contents of the string literal after processing escape sequences.

Reply:
> I think this is better implemented as a for loop

Yup, the suggested for loop reads better and is more concise. Just curious, is there any other reason you suggested this refactor?

> Also note the use of cooked, which is the contents of the string literal after processing escape sequences.

Good to know - thanks for the info!